File: examples/python/02_medium_inject.py
"""02 Medium: Programmatic WHERE injection — QQL's superpower."""
import pyqql

q = "QUERY 'machine learning transformer' FROM papers LIMIT 20"

s = pyqql.inject_filter(q, "tenant_id", "=", "acme-corp")
print("=== String filter ===")
print(s[:400])

s = pyqql.inject_filter(q, "impact_factor", ">=", 5.0)
print("\n=== Numeric filter ===")
print(s[:400])

s = pyqql.inject_filter(q, "is_published", "=", True)
print("\n=== Boolean filter ===")
print(s[:400])

File: examples/nodejs/02_medium_inject.mjs
// Medium: Programmatic WHERE injection — apply security filters to queries.
import { injectFilter, isValid } from 'nqql';

const userQuery = "QUERY 'machine learning transformer' FROM papers LIMIT 20";
console.log(`User query valid: ${isValid(userQuery)}`);

// Inject a tenant_id filter (string value)
const tenantQuery = injectFilter(
  userQuery, 'tenant_id', '=', '{"str": "acme-corp"}',
);
console.log('\n=== Tenant isolation ===');
console.log(tenantQuery.substring(0, 500));

// Inject a numeric threshold
const boosted = injectFilter(
  userQuery, 'impact_factor', '>=', '{"float": 5.0}',
);
console.log('\n=== Numeric threshold ===');
console.log(boosted.substring(0, 500));

// Inject a boolean flag
const published = injectFilter(
  userQuery, 'is_published', '=', '{"bool": true}',
);
console.log('\n=== Boolean filter ===');
console.log(published.substring(0, 500));
